tahrir_api/dbapi.py

```diff
@@ -761,12 +761,23 @@ def get_all_issuers(self):
 
         return self.session.query(Issuer)
 
-    def get_all_assertions(self):
+    def get_all_assertions(self, begin: int = 0, limit: int = 100):
         """
-        Get all assertions in the db.
+        Get all assertions in the db, ordered by most recent first.
+
+        :type begin: int
+        :param begin: Number of assertions to skip (offset for pagination, default: 0)
+
+        :type limit: int
+        :param limit: Maximum number of assertions to return (default: 100)
         """
 
-        return self.session.query(Assertion)
+        return (
+            self.session.query(Assertion)
+            .order_by(Assertion.issued_on.desc())
```

Comment:
You're right to sort on something when limiting results, and `issued_on` looks like a good candidate, however it is not indexed in the DB, so the sorting may take quite some time on large datasets. Please add an index (and the associated alembic migration).